ibm_is_floating_ip nil pointer issue when target is not set

<!---
Please note the following potential times when an issue might be in Terraform core:

* [Configuration Language](https://www.terraform.io/docs/configuration/index.html) or resource ordering issues
* [State](https://www.terraform.io/docs/state/index.html) and [State Backend](https://www.terraform.io/docs/backends/index.html) issues
* [Provisioner](https://www.terraform.io/docs/provisioners/index.html) issues
* [Registry](https://registry.terraform.io/) issues
* Spans resources across multiple providers

If you are running into one of these scenarios, we recommend opening an issue in the [Terraform core repository](https://github.com/hashicorp/terraform/) instead.
--->

<!--- Please keep this note for the community --->

<!--- Thank you for keeping this note for the community --->

### Terraform CLI and Terraform IBM Provider Version 1.41.0

<!--- Please run `terraform -v` to show the Terraform core version and provider version(s). If you are not running the latest version of Terraform or the provider, please upgrade because your issue may have already been fixed. [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions). --->

### Affected Resource(s)

<!--- Please list the affected resources and data sources. --->

* ibm_is_floating_ip

### Terraform Configuration Files

<!--- Information about code formatting: https://help.github.com/articles/basic-writing-and-formatting-syntax/#quoting-code --->

Please include all Terraform configurations required to reproduce the bug. Bug reports without a functional reproduction may be closed without investigation.

```hcl
resource ibm_is_floating_ip this {
  name = "fip-1"
  zone 				= "us-south-1"
}
resource "ibm_is_bare_metal_server_network_interface_floating_ip" "testacc_bms_fip" {
  bare_metal_server = ibm_is_bare_metal_server.testacc_bms.id
  network_interface = ibm_is_bare_metal_server.testacc_bms.primary_network_interface.0.id
  floating_ip       = ibm_is_floating_ip.this.id
}
```

### Debug Output

<!---
Please provide a link to a GitHub Gist containing the complete debug output. Please do NOT paste the debug output in the issue; just paste a link to the Gist.

To obtain the debug output, see the [Terraform documentation on debugging](https://www.terraform.io/docs/internals/debugging.html).
--->

### Panic Output
```
Stack trace from the terraform-provider-ibm_v1.41.0 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
```
<!--- If Terraform produced a panic, please provide a link to a GitHub Gist containing the output of the `crash.log`. --->

### Expected Behavior

<!--- What should have happened? --->

### Actual Behavior

<!--- What actually happened? --->

### Steps to Reproduce

<!--- Please list the steps required to reproduce the issue. --->

1. `terraform apply`

### Important Factoids

<!--- Are there anything atypical about your accounts that we should know? For example: Running in EC2 Classic? --->

### References

<!---
Information about referencing Github Issues: https://help.github.com/articles/basic-writing-and-formatting-syntax/#referencing-issues-and-pull-requests

Are there any other GitHub issues (open or closed) or pull requests that should be linked here? Vendor documentation? For example:
--->

* #0000

--- a/ibm/service/vpc/resource_ibm_is_floating_ip.go
+++ b/ibm/service/vpc/resource_ibm_is_floating_ip.go
@@ -107,6 +107,7 @@ func ResourceIBMISFloatingIP() *schema.Resource {
 				Optional:      true,
 				Computed:      true,
 				ConflictsWith: []string{isFloatingIPTarget},
+				ExactlyOneOf:  []string{isFloatingIPTarget, isFloatingIPZone},
 				Description:   "Zone name",
 			},
 
@@ -115,6 +116,7 @@ func ResourceIBMISFloatingIP() *schema.Resource {
 				Optional:      true,
 				Computed:      true,
 				ConflictsWith: []string{isFloatingIPZone},
+				ExactlyOneOf:  []string{isFloatingIPTarget, isFloatingIPZone},
 				Description:   "Target info",
 			},
 			floatingIPTargets: {
@@ -396,7 +398,7 @@ func fipGet(d *schema.ResourceData, meta interface{}, id string) error {
 	} else {
 		d.Set(isFloatingIPTarget, "")
 	}
-	targetList := []map[string]interface{}{}
+	targetList := make([]map[string]interface{}, 0)
 	targetList = append(targetList, targetMap)
 	d.Set(floatingIPTargets, targetList)
 	tags, err := flex.GetTagsUsingCRN(meta, *floatingip.CRN)
@@ -660,122 +662,124 @@ func checkIfZoneChanged(oldNic, newNic, currentZone string, floatingipC *vpcv1.V
 func floatingIPCollectionFloatingIpTargetToMap(targetItemIntf vpcv1.FloatingIPTargetIntf) (targetId string, targetMap map[string]interface{}) {
 	targetMap = map[string]interface{}{}
 	targetId = ""
-	switch reflect.TypeOf(targetItemIntf).String() {
-	case "*vpcv1.FloatingIPTargetNetworkInterfaceReference":
-		{
-			targetItem := targetItemIntf.(*vpcv1.FloatingIPTargetNetworkInterfaceReference)
-			targetId = *targetItem.ID
-			if targetItem.Deleted != nil {
-				deletedList := []map[string]interface{}{}
-				deletedMap := floatingIPTargetNicDeletedToMap(*targetItem.Deleted)
-				deletedList = append(deletedList, deletedMap)
-				targetMap[floatingIPTargetsDeleted] = deletedList
-			}
-			if targetItem.Href != nil {
-				targetMap[floatingIPTargetsHref] = targetItem.Href
-			}
-			if targetItem.ID != nil {
-				targetMap[floatingIPTargetsId] = targetItem.ID
-			}
-			if targetItem.Name != nil {
-				targetMap[floatingIPTargetsName] = targetItem.Name
-			}
-			if targetItem.PrimaryIP != nil {
-				primaryIpList := make([]map[string]interface{}, 0)
-				currentIP := map[string]interface{}{}
-				if targetItem.PrimaryIP.Address != nil {
-					currentIP[floatingIpPrimaryIpAddress] = *targetItem.PrimaryIP.Address
+	if targetItemIntf != nil {
+		switch reflect.TypeOf(targetItemIntf).String() {
+		case "*vpcv1.FloatingIPTargetNetworkInterfaceReference":
+			{
+				targetItem := targetItemIntf.(*vpcv1.FloatingIPTargetNetworkInterfaceReference)
+				targetId = *targetItem.ID
+				if targetItem.Deleted != nil {
+					deletedList := []map[string]interface{}{}
+					deletedMap := floatingIPTargetNicDeletedToMap(*targetItem.Deleted)
+					deletedList = append(deletedList, deletedMap)
+					targetMap[floatingIPTargetsDeleted] = deletedList
 				}
-				if targetItem.PrimaryIP.Href != nil {
-					currentIP[floatingIpPrimaryIpHref] = *targetItem.PrimaryIP.Href
+				if targetItem.Href != nil {
+					targetMap[floatingIPTargetsHref] = targetItem.Href
 				}
-				if targetItem.PrimaryIP.Name != nil {
-					currentIP[floatingIpPrimaryIpName] = *targetItem.PrimaryIP.Name
+				if targetItem.ID != nil {
+					targetMap[floatingIPTargetsId] = targetItem.ID
 				}
-				if targetItem.PrimaryIP.ID != nil {
-					currentIP[floatingIpPrimaryIpId] = *targetItem.PrimaryIP.ID
+				if targetItem.Name != nil {
+					targetMap[floatingIPTargetsName] = targetItem.Name
 				}
-				if targetItem.PrimaryIP.ResourceType != nil {
-					currentIP[floatingIpPrimaryIpResourceType] = *targetItem.PrimaryIP.ResourceType
+				if targetItem.PrimaryIP != nil {
+					primaryIpList := make([]map[string]interface{}, 0)
+					currentIP := map[string]interface{}{}
+					if targetItem.PrimaryIP.Address != nil {
+						currentIP[floatingIpPrimaryIpAddress] = *targetItem.PrimaryIP.Address
+					}
+					if targetItem.PrimaryIP.Href != nil {
+						currentIP[floatingIpPrimaryIpHref] = *targetItem.PrimaryIP.Href
+					}
+					if targetItem.PrimaryIP.Name != nil {
+						currentIP[floatingIpPrimaryIpName] = *targetItem.PrimaryIP.Name
+					}
+					if targetItem.PrimaryIP.ID != nil {
+						currentIP[floatingIpPrimaryIpId] = *targetItem.PrimaryIP.ID
+					}
+					if targetItem.PrimaryIP.ResourceType != nil {
+						currentIP[floatingIpPrimaryIpResourceType] = *targetItem.PrimaryIP.ResourceType
+					}
+					primaryIpList = append(primaryIpList, currentIP)
+					targetMap[floatingIpPrimaryIP] = primaryIpList
 				}
-				primaryIpList = append(primaryIpList, currentIP)
-				targetMap[floatingIpPrimaryIP] = primaryIpList
-			}
-			if targetItem.ResourceType != nil {
-				targetMap[floatingIPTargetsResourceType] = targetItem.ResourceType
-			}
-		}
-	case "*vpcv1.FloatingIPTargetPublicGatewayReference":
-		{
-			targetItem := targetItemIntf.(*vpcv1.FloatingIPTargetPublicGatewayReference)
-			targetId = *targetItem.ID
-			if targetItem.Deleted != nil {
-				deletedList := []map[string]interface{}{}
-				deletedMap := floatingIPTargetPgDeletedToMap(*targetItem.Deleted)
-				deletedList = append(deletedList, deletedMap)
-				targetMap[floatingIPTargetsDeleted] = deletedList
-			}
-			if targetItem.Href != nil {
-				targetMap[floatingIPTargetsHref] = targetItem.Href
-			}
-			if targetItem.ID != nil {
-				targetMap[floatingIPTargetsId] = targetItem.ID
-			}
-			if targetItem.Name != nil {
-				targetMap[floatingIPTargetsName] = targetItem.Name
-			}
-			if targetItem.ResourceType != nil {
-				targetMap[floatingIPTargetsResourceType] = targetItem.ResourceType
-			}
-			if targetItem.CRN != nil {
-				targetMap[floatingIPTargetsCrn] = targetItem.CRN
-			}
-		}
-	case "*vpcv1.FloatingIPTarget":
-		{
-			targetItem := targetItemIntf.(*vpcv1.FloatingIPTarget)
-			targetId = *targetItem.ID
-			if targetItem.Deleted != nil {
-				deletedList := []map[string]interface{}{}
-				deletedMap := floatingIPTargetNicDeletedToMap(*targetItem.Deleted)
-				deletedList = append(deletedList, deletedMap)
-				targetMap[floatingIPTargetsDeleted] = deletedList
-			}
-			if targetItem.Href != nil {
-				targetMap[floatingIPTargetsHref] = targetItem.Href
-			}
-			if targetItem.ID != nil {
-				targetMap[floatingIPTargetsId] = targetItem.ID
-			}
-			if targetItem.Name != nil {
-				targetMap[floatingIPTargetsName] = targetItem.Name
-			}
-			if targetItem.PrimaryIP != nil && targetItem.PrimaryIP.Address != nil {
-				primaryIpList := make([]map[string]interface{}, 0)
-				currentIP := map[string]interface{}{}
-				if targetItem.PrimaryIP.Address != nil {
-					currentIP[floatingIpPrimaryIpAddress] = *targetItem.PrimaryIP.Address
+				if targetItem.ResourceType != nil {
+					targetMap[floatingIPTargetsResourceType] = targetItem.ResourceType
 				}
-				if targetItem.PrimaryIP.Href != nil {
-					currentIP[floatingIpPrimaryIpHref] = *targetItem.PrimaryIP.Href
-				}
-				if targetItem.PrimaryIP.Name != nil {
-					currentIP[floatingIpPrimaryIpName] = *targetItem.PrimaryIP.Name
-				}
-				if targetItem.PrimaryIP.ID != nil {
-					currentIP[floatingIpPrimaryIpId] = *targetItem.PrimaryIP.ID
-				}
-				if targetItem.PrimaryIP.ResourceType != nil {
-					currentIP[floatingIpPrimaryIpResourceType] = *targetItem.PrimaryIP.ResourceType
-				}
-				primaryIpList = append(primaryIpList, currentIP)
-				targetMap[floatingIpPrimaryIP] = primaryIpList
 			}
-			if targetItem.ResourceType != nil {
-				targetMap[floatingIPTargetsResourceType] = targetItem.ResourceType
+		case "*vpcv1.FloatingIPTargetPublicGatewayReference":
+			{
+				targetItem := targetItemIntf.(*vpcv1.FloatingIPTargetPublicGatewayReference)
+				targetId = *targetItem.ID
+				if targetItem.Deleted != nil {
+					deletedList := []map[string]interface{}{}
+					deletedMap := floatingIPTargetPgDeletedToMap(*targetItem.Deleted)
+					deletedList = append(deletedList, deletedMap)
+					targetMap[floatingIPTargetsDeleted] = deletedList
+				}
+				if targetItem.Href != nil {
+					targetMap[floatingIPTargetsHref] = targetItem.Href
+				}
+				if targetItem.ID != nil {
+					targetMap[floatingIPTargetsId] = targetItem.ID
+				}
+				if targetItem.Name != nil {
+					targetMap[floatingIPTargetsName] = targetItem.Name
+				}
+				if targetItem.ResourceType != nil {
+					targetMap[floatingIPTargetsResourceType] = targetItem.ResourceType
+				}
+				if targetItem.CRN != nil {
+					targetMap[floatingIPTargetsCrn] = targetItem.CRN
+				}
 			}
-			if targetItem.CRN != nil {
-				targetMap[floatingIPTargetsCrn] = targetItem.CRN
+		case "*vpcv1.FloatingIPTarget":
+			{
+				targetItem := targetItemIntf.(*vpcv1.FloatingIPTarget)
+				targetId = *targetItem.ID
+				if targetItem.Deleted != nil {
+					deletedList := []map[string]interface{}{}
+					deletedMap := floatingIPTargetNicDeletedToMap(*targetItem.Deleted)
+					deletedList = append(deletedList, deletedMap)
+					targetMap[floatingIPTargetsDeleted] = deletedList
+				}
+				if targetItem.Href != nil {
+					targetMap[floatingIPTargetsHref] = targetItem.Href
+				}
+				if targetItem.ID != nil {
+					targetMap[floatingIPTargetsId] = targetItem.ID
+				}
+				if targetItem.Name != nil {
+					targetMap[floatingIPTargetsName] = targetItem.Name
+				}
+				if targetItem.PrimaryIP != nil && targetItem.PrimaryIP.Address != nil {
+					primaryIpList := make([]map[string]interface{}, 0)
+					currentIP := map[string]interface{}{}
+					if targetItem.PrimaryIP.Address != nil {
+						currentIP[floatingIpPrimaryIpAddress] = *targetItem.PrimaryIP.Address
+					}
+					if targetItem.PrimaryIP.Href != nil {
+						currentIP[floatingIpPrimaryIpHref] = *targetItem.PrimaryIP.Href
+					}
+					if targetItem.PrimaryIP.Name != nil {
+						currentIP[floatingIpPrimaryIpName] = *targetItem.PrimaryIP.Name
+					}
+					if targetItem.PrimaryIP.ID != nil {
+						currentIP[floatingIpPrimaryIpId] = *targetItem.PrimaryIP.ID
+					}
+					if targetItem.PrimaryIP.ResourceType != nil {
+						currentIP[floatingIpPrimaryIpResourceType] = *targetItem.PrimaryIP.ResourceType
+					}
+					primaryIpList = append(primaryIpList, currentIP)
+					targetMap[floatingIpPrimaryIP] = primaryIpList
+				}
+				if targetItem.ResourceType != nil {
+					targetMap[floatingIPTargetsResourceType] = targetItem.ResourceType
+				}
+				if targetItem.CRN != nil {
+					targetMap[floatingIPTargetsCrn] = targetItem.CRN
+				}
 			}
 		}
 	}

